### ISSUE TITLE
Port twisted.python.reflect.prefixedMethodNames to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6100]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#6100|
|Type|enhancement|
|Created|2012-10-11 19:43:10Z|
|Branch|https://github.com/twisted/twisted/tree/prefixedmethodnames-py3-6100|

This is needed by `twisted.web.resource`.

<details><summary>Searchable metadata</summary>

```
trac-id__6100 6100
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__Python_3_3_Minimal Python 3.3 Minimal
branch__branches_prefixedmethodnames_py3_6100 branches/prefixedmethodnames-py3-6100
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__web web
keywords__ 
time__1349984590000000 1349984590000000
changetime__1350144683000000 1350144683000000
version__None None
owner__exarkun exarkun
cc__jknight
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [36105]) Branching to 'prefixedmethodnames-py3-6100'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

(In [36106]) Move prefixedMethodNames and helper to _reflectpy3.py and port it to Python 3; add some tests for it, since it had none.

refs [#6100](https://github.com/twisted/twisted/issues/6100)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set owner to @itamarst|
|-|-|

Easy, _almost_ just a move.  [Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/prefixedmethodnames-py3-6100)

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_5"></a>|@itamarst set owner to @exarkun|
|-|-|

  1. `test_inheritedMethod` isn't actually checking methods on `Child`.
  1. `addMethodNamesToDict` is horrible - new ticket?

Otherwise good, fix then merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

(In [36140]) Fix new test to use child class

refs [#6100](https://github.com/twisted/twisted/issues/6100)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set status to `closed`|
|-|-|

(In [36141]) Merge prefixedmethodnames-py3-6100

Author: exarkun
Reviewer: itamarst
Fixes: [#6100](https://github.com/twisted/twisted/issues/6100)

Port `prefixedMethodNames` to Python 3.
